tests/plugins/widget/deleteparagraphsclosetowidgets.js

```diff
@@ -0,0 +1,193 @@
+/* bender-tags: editor */
+/* bender-ckeditor-plugins: wysiwygarea,toolbar,forms,widget,codesnippet,undo,basicstyles,entities */
+
+'use strict';
+( function() {
+
+	var backspace = 8,
+		deleteKey = 46,
+		leftArrow = 37,
+		rightArrow = 39;
+
+	var input = (
+		'<div class="simplebox">' +
+		'<h2 class="simplebox-title">Title</h2>' +
+		'<div class="simplebox-content"><p>Content...</p></div></div>' +
+		'<p>&nbsp;</p>' +
+		'<pre><code class="language-javascript">//Hello another world!</code></pre>'
+	);
+
+	var notEmptyInput = (
+		'<div class="simplebox">' +
+		'<h2 class="simplebox-title">Title</h2>' +
+		'<div class="simplebox-content"><p>Content...</p></div></div>' +
+		'<p>Hello, World!</p>' +
+		'<pre><code class="language-javascript">//Hello another world!</code></pre>'
+	);
+
+	var result = (
+		'<div class="simplebox">' +
+		'<h2 class="simplebox-title">Title</h2>' +
+		'<div class="simplebox-content"><p>Content...</p></div></div>' +
+		'<pre><code class="language-javascript">//Hello another world!</code></pre>'
+	);
+
+	var simpleBox = {
+		template: '<div class="simplebox">' +
+			'<h2 class="simplebox-title">Title</h2>' +
+			'<div class="simplebox-content"><p>Content...</p></div>' +
+		'</div>',
+		editables: {
+			title: {
+				selector: '.simplebox-title',
+				allowedContent: 'br strong em'
+			},
+			content: {
+				selector: '.simplebox-content',
+				allowedContent: 'p br ul ol li strong em'
+			}
+		},
+		allowedContent: 'div(!simplebox); div(!simplebox-content); h2(!simplebox-title)',
+		requiredContent: 'div(simplebox)',
+
+		upcast: function( element ) {
+			return element.name === 'div' && element.hasClass( 'simplebox' );
+		}
+	};
+
+	function selectEmptyParagraph( editor ) {
+		var range = editor.createRange();
+
+		range.moveToPosition( editor.document.getElementsByTag( 'P' ).getItem( 1 ), CKEDITOR.POSITION_AFTER_START );
+		editor.getSelection().selectRanges( [ range ] );
+	}
+
+	bender.test( {
+
+		// (#1572)
+		'test fire delete to remove an empty paragraph between two widgets': function() {
```

Comment:
Tests 1-4 and 5-6 have very similar structure. Could you refactor them to extract the common logic? For inspiration you can check https://github.com/ckeditor/ckeditor4/blob/major/tests/plugins/colorbutton/ariaattributes.js.